okhttp/src/main/java/okhttp3/internal/http/HttpEngine.java

```diff
@@ -499,6 +500,13 @@ public static boolean hasBody(Response response) {
    * content types the application is interested in.
    */
   private Request networkRequest(Request request) throws IOException {
+    if (!request.isHttps()) {
+      Authenticator authenticator = client.proxyAuthenticator();
```

Comment:
@yschimke I only need it for proxy. Did not bother think about regular authenticator. I like @swankjesse suggestion about the setting in the client. I will implement it.